### DISCUSSION TITLE
GET Verifiable Credentials in json format from an external wallet

### DISCUSSION THREAD

**Discussion by santidediego** [eclipse-edc/Connector#4352] [Q&A]
I am wondering how I can incorporate a VC in the connector from an external wallet. At startup, the connector does an HTTP GET request against and endpoint and receives a .json file containing a VC, which would be presented later by the connector when requested. **Any thoughts about if there are already some extensions that allow me to fulfil this?**

For the moment I've seen that there is an extension for http under [http/](https://github.com/eclipse-edc/Connector/tree/main/extensions/common/http) and a group of extensions for working with decentralized identity and verifiable credentials under [iam/](https://github.com/eclipse-edc/Connector/tree/main/extensions/common/iam).

By looking at the code of the SSI extension of the connector, I see that at one point it has a method _getCredentialServiceClient_, in [this file](https://github.com/eclipse-edc/Connector/blob/main/extensions/common/iam/identity-trust/identity-trust-core/src/main/java/org/eclipse/edc/iam/identitytrust/core/IdentityAndTrustExtension.java) but I do not understand what it does. I can see that internally it uses a variable of type _EdcHttpClient_ a imports methods from the library _org.eclipse.edc.http.spi.EdcHttpClient;_

Thanks in advance.

**Comment by paullatzelsperger** [comment]
hi @santidediego I STRONGLY recommend getting familiar with core concepts like DSP (Dataspace Protocol), DCP (Decentralized Claims Protocol) and once you've fully grokked that, take a look at our CredentialService reference implementation called IdentityHub.
In addition, there is our MinimumViableDataspace repo, that should get you started.
